backend/spec/performance/invoice_list_query_performance_spec.rb

```
@@ -0,0 +1,170 @@
+        efficient_time = measure_query_time(efficient_query)
+        
+        # The efficient query should be faster (though difference may be small with test data)
+        expect(efficient_time).to be <= inefficient_time
+        
+        # Both should complete reasonably quickly with test data
+        expect(efficient_time).to be < 100.0 # 100ms threshold for test environment
+      end
+    end
+  end
+
+  private
```

RuboCop (1.75.5)

[convention] Remove a blank line after `private`.

(Layout/EmptyLinesAroundAccessModifier)